### PR TITLE
[engine] changes to DlVertices::Builder and Stopwatch visualizer.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Collection of changes to DlVertices::Builder and the stopwatch visualizer.

At a high level:

* improve performance of the stopwatch visualizer by pre-allocating storage (and sharing it across both visualizers), lookup up font once, and cache the debug frame rate used. Updates to use Dl types instead of SkTypes.

* Change DlVerticesBuilder to allow storing the bounds and use that in the visualizer, since we already know them. Make FML_CHECKS into dchecks, as the dart:ui vertices will already bounds check correctly - so these should only be necessary for debugging engine changes.

### CODE REVIEW

File: flow/layers/performance_overlay_layer.cc [hunk 1]
```diff
@@ -134,16 +137,22 @@ void PerformanceOverlayLayer::Paint(PaintContext& context) const {
   SkScalar width = paint_bounds().width() - (padding * 2);
   SkScalar height = paint_bounds().height() / 2;
   auto mutator = context.state_stack.save();
+  // Cached storage for vertex output.
+  std::vector<DlPoint> vertices_storage;
+  std::vector<DlColor> color_storage;
```
Comment (flar) [inline flow/layers/performance_overlay_layer.cc:142]: Weren't these supposed to be cached somewhere? It looks like they are recreated on every frame?

Comment (flar) [reply flow/layers/performance_overlay_layer.cc:142]: Or is this just re-use between the 2 graphs?

Comment (jonahwilliams) [reply flow/layers/performance_overlay_layer.cc:142]: Just between the two graphs. originally I was much more ambitious but it turns out that we rebuild the layer each frame so there isn't a good place to put the storage.

File: flow/stopwatch_dl.cc [hunk 2]
```diff
@@ -106,48 +115,50 @@ void DlStopwatchVisualizer::Visualize(DlCanvas* canvas,
...
-      SkPoint::Make(right, bottom),  //
-      SkPoint::Make(right, bottom),  // tl
-      SkPoint::Make(left, bottom),   // bl br
-      SkPoint::Make(left, top)       //
+  auto const left = rect.GetLeft();
+  auto const top = rect.GetTop();
+  auto const right = rect.GetRight();
+  auto const bottom = rect.GetBottom();
+
+  auto const vertices = std::array<DlPoint, 6>{
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply flow/stopwatch_dl.cc:133]: Yeah good point, let me swap that out.

[the remark this answers is not in the document]
Comment (jonahwilliams) [reply flow/stopwatch_dl.cc]: Done

**Review by flar — APPROVED** [review]
LGTM

**Pull request merged**